server/src/main/java/org/elasticsearch/search/vectors/KnnSearchBuilder.java

```diff
@@ -70,6 +70,25 @@ public static KnnSearchBuilder fromXContent(XContentParser parser) throws IOExce
         return PARSER.parse(parser, null);
     }
 
+    public static List<KnnSearchBuilder> fromXContentList(XContentParser parser) throws IOException {
+        List<KnnSearchBuilder> knnSearchBuilders = new ArrayList<>();
+        XContentParser.Token token = parser.currentToken();
+        if (token == XContentParser.Token.START_ARRAY) {
+            while ((token = parser.nextToken()) != XContentParser.Token.END_ARRAY) {
+                if (token == XContentParser.Token.START_OBJECT) {
+                    knnSearchBuilders.add(fromXContent(parser));
+                } else {
+                    throw new IllegalArgumentException("malformed knn format, within the knn search array only objects are allowed");
```

Comment:
you could use XContentParseException here, which can incorporate the line taken from the parser for better error reporting. Also shall we print out which unexpected token was found?